packages/lib/delivery/src/Notification.ts

```diff
@@ -222,6 +222,54 @@ export const verifyOtp = async (
     db.resetOtp(ensName, notificationChannelType);
 };
 
+// method to enable/disable notification channel
+export const enableOrDisableNotificationChannel = async (
+    ensName: string,
+    isEnabled: boolean,
+    notificationChannelType: NotificationChannelType,
+    dsNotificationChannels: NotificationChannel[],
+    db: any,
+) => {
+    // check if channel is supported or not
+    const channelUsed = dsNotificationChannels.filter(
+        (channel) => channel.type === notificationChannelType,
+    );
+
+    if (!channelUsed.length) {
+        throw new NotificationError(
+            `Notification channel ${notificationChannelType} is currently not supported by the DS`,
+        );
+    }
+
+    // check if notification channel exists in DB
+    const userNotificationChannels: NotificationChannel[] =
+        await db.getUsersNotificationChannels(ensName);
+
+    const channelToUpdate = userNotificationChannels.filter(
+        (data) => data.type === notificationChannelType,
+    );
+
+    if (!channelToUpdate.length) {
+        throw Error(
+            `${notificationChannelType} notification channel is not configured`,
+        );
+    }
+
+    // check if notification channel is alaready enabled/disabled
+    checkAlreadyEnabledOrDisabled(
```

Comment:
I think we don't need to throw an error here. If the user intends to activate the channel that is already active we can simply return 200 because his intent is fullfiled anyways

Reply:
ok